### ISSUE TITLE
Cant' start NDS (v1 or v2): br-lan not detected

### ISSUE THREAD

**Issue by deltabi** [nodogsplash/nodogsplash#233]
Hello,
I'm configuring a GL.iNet AR300M and I would like to add NDS for a simple captive portal.
I'm using the "clean" firmware from GL.iNet which is based on OpenWRT 15.05. The repository NDS package provided by GLi is version 0.9Beta, this version works correctly but I need the ipset functionality which has been added in v1 (if I understood correctly). Using the opkg command I upgraded the beta version to the v1.0.2 downloaded from the snapshot , but when I start NDS it shows the following message: "nodogsplash: br-lan not detected, NoDogSplash not starting". The same happens with version v2.0.0-1.
What's wrong? Interface br-lan is present on the device (it bridges LAN port with a Wifi network).
Thanks.

**Comment by bluewavenet** [comment]
First, I woud go for the latest LEDE/OpenWrt firmware at:
https://downloads.lede-project.org/releases/17.01.4/targets/ar71xx/generic/lede-17.01.4-ar71xx-generic-gl-ar300m-squashfs-sysupgrade.bin

Then get NDS 2.0.1 from snapshot at:
https://downloads.lede-project.org/snapshots/packages/mips_24kc/routing/nodogsplash2_2.0.1-1_mips_24kc.ipk
This version fixes a bug that causes your problem where the lan interface has not come up fully before NDS tries to start.
Get both of these as described then report back. I think this combination will just work without any addtional additional configuration orher than turning on the wifi.

**Comment by deltabi** [comment]
Thanks for the response.
Got it working with the linked firmware/package.
However I can't configure an ipset rule within NDS.  I have an ipset called "arogis" and I want to allow access to those IPs to the "preauthenticated_users".
I've used the following syntax without success:
  list preauthenticated_users 'allow ipset arogis'
  list preauthenticated_users 'allow to ipset arogis'
  list preauthenticated_users 'allow to-ipset arogis'
A syntax error is shown in system log in all those cases.
What's the correct syntax for the ipset rule in the NDS configuration file (the one in /etc/config)?

**Comment by bluewavenet** [comment]
I think  your first one is correct ie
 list preauthenticated_users 'allow ipset arogis'
although I have not used ipset support to try it out.
Is there any more in the log other than syntax error?

Once NDS is started there will be a file:
/tmp/etc/nodogsplash_xxxxx.conf
where xxxxx is some string.
Does your ipset rule appear in there?

**Comment by deltabi** [comment]
Retested the first syntax ('allow ipset arogis') after a reboot and it looks like it's working... probably I messed up something in the previous tests as I had the impressions the firewall rules where not in place correctly (I restarted NDS, fw and dnsmasq several times in different order for configuring the ipset).
Anyway it appears to work correctly now and the /tmp/etc/nodogsplash_xxxxx.conf contains the relevant 'FirewallRule allow ipset' statement.
The case can be closed. Thank you.
